File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let ans = '';
    let minLength = Infinity;
    for (let str of strs) {
        minLength = Math.min(minLength, str.length);
    }
    
    let i = 0;
    while (i < minLength) {
        let ch = strs[0][i];
        if (strs.map(s => s[i]).every(s => s === ch)) {
            ans += ch;
            i++;
        } else {
            break;
        }
    }
    
    return ans;
};

File: 14-longest-common-prefix/14-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    ans = ''
    minLength = strs.map{ |s| s.length }.min
    
    (0...minLength).each do |i| 
        ch = strs[0][i]
        if strs.all? { |s| s[i] == ch }
            ans += ch
        else
            break
        end
    end
    
    ans
end